### DISCUSSION TITLE
[Feature Request] A modhook for hit detection

### DISCUSSION THREAD

**Discussion by TheReal-Kaze** [SmartlyDressedGames/Unturned-3.x-Community#4546] [Ideas]
Here is the working explained in few lines:
-So instead of using the rubble system we can use that new modhook and for example to not complicate things let’s keep in mid the working of the old rubble system, we have a main parent “Sections” ,then “Section_1” but instead of having alive and dead, we put the script on the main Gameobject “Sections” and then all the “Section_x” will be considered as a different part of the object and will have its own life then when it get destroyed it does just disable “Section_x” and instead of having the reset timer hard coded just leave modders use timer even hook combined to activation hook to have the same result with the maximum of flexibility. This will fix the timer problems the rubble system, and my idea can even be more improved to have the most optimal thingy ever 

Why and what will it add to unturned?
-First of all you will start to see real mechanics in unturned like shooting ranges etc or even walls that can be destroyed realistically like in rainbow six siege, and maybe even make the script can be used on barricades , { for blade ID it can be added later too }

**Comment by GazziFX** [comment]
We already have quite a large number of modhooks and adding even more for edge cases is a bad idea. Therefore, it is better to spend efforts on a more universal solution such as client-side scripting language #3930
